moto/iot/models.py

```diff
@@ -374,6 +374,61 @@ def to_dict(self):
         return obj
 
 
+class FakeEndpoint(BaseModel):
+    def __init__(self, endpoint_type, region_name):
+        if endpoint_type not in [
+            "iot:Data",
+            "iot:Data-ATS",
+            "iot:CredentialProvider",
+            "iot:Jobs",
+        ]:
+            raise InvalidRequestException(
+                " An error occurred (InvalidRequestException) when calling the DescribeEndpoint "
+                "operation: Endpoint type %s not recognized." % endpoint_type
+            )
+        self.region_name = region_name
+        data_identifier = "".join(
```

Comment:
There's already a `_random_string` method in IoTBackend - would be good to extract that into a utils.py, and reuse it here

Reply:
sure thing